File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String input = br.readLine();

        System.out.println(solution(input));
    }

    public static int solution(String input) {
        int currentIndex = 0;
        int num = 1;

        while (currentIndex < input.length()) {
            String numStr = Integer.toString(num);

            for (char c : numStr.toCharArray()) {
                if(c == input.charAt(currentIndex)) {
                    currentIndex++;
                    if (currentIndex >= input.length()) {
                        return num;
                    }
                }
            }

            num++;
        }

        return num;
    }
}

File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.swift
//
//  main.swift
//  BJ1515
//
//  Created by 안홍범 on 4/10/25.
//

import Foundation

let input = readLine()!

func solution(_ input: String) -> Int {
    var currentIndex = 0
    var num = 1
    let inputArray = Array(input)

    while currentIndex < inputArray.count {
        let numStr = String(num)
        
        for char in numStr {
            if char == inputArray[currentIndex] {
                currentIndex += 1
                if currentIndex >= inputArray.count {
                    return num
                }
            }
        }
        
        num += 1
    }
    
    return num
}

print(solution(input))
